src/metabase/lib/query.cljc

```diff
@@ -29,28 +86,15 @@
      ;; `location` should probably be "found" first before iterating.
      (reduce
        (fn [query location]
-         (lib.util/update-query-stage query stage-number
-                                      lib.util/replace-clause location target-clause replacement))
+         (let [result (lib.util/update-query-stage query stage-number
+                                                   lib.util/replace-clause location target-clause replacement)]
```

Comment:
This seems to be the only difference from `remove-clause`, so it might make sense to have a shared implementation.

Reply:
I want to implement more removal targets first, but I'll keep an eye open.